plugin/core/file_watcher.py

```diff
@@ -0,0 +1,76 @@
+from .typing import List, Optional, Protocol, Tuple, Type
+from abc import ABCMeta
+from abc import abstractmethod
+
+
+DEFAULT_IGNORES = ['**/.git/**', '**/node_modules/**', '**/.hg/**']
```

Comment:
Would be better if this were binary_file_patterns and/or index_exclude_patterns. But the glob syntax/features of ST is unfortunately incomplete compared to VSCode.

Reply:
Wouldn't you need `**/.svn/**` here as well?

Reply:
Someone is using SVN still? :)

I can add of course but was planning too look into fetching ignores from ST and converting to glob but that might turn out to be some work.

Reply:
> Someone is using SVN still? :)

Unfortunately, yes, these companies exist ;) But I think it's okay to forget about it and focus on fetching the ignore list from ST instead.